tests/strands/experimental/test_agent_config.py

```diff
@@ -0,0 +1,271 @@
+"""Tests for experimental AgentConfig."""
+
+import pytest
+from unittest.mock import Mock, patch
+
+from strands.experimental.agent_config import AgentConfig
+from strands.experimental.tool_pool import ToolPool
+from strands.types.tools import AgentTool
+
+
+class TestAgentConfig:
+    """Test experimental AgentConfig functionality."""
+    
+    def test_agent_config_creation(self):
+        """Test AgentConfig can be created with dict config."""
+        # Provide empty ToolPool since strands_tools not available in tests
+        config = AgentConfig({"model": "test-model"}, tool_pool=ToolPool())
+        assert config.model == "test-model"
+    
+    def test_agent_config_with_tools(self):
+        """Test AgentConfig with basic configuration."""
+        
+        config = AgentConfig({
+            "model": "test-model",
+            "prompt": "Test prompt"
+        }, tool_pool=ToolPool())
+        
+        assert config.model == "test-model"
+        assert config.system_prompt == "Test prompt"
+    
+    def test_agent_config_file_prefix_required(self):
+        """Test that file paths must have file:// prefix."""
+        
+        with pytest.raises(ValueError, match="File paths must be prefixed with 'file://'"):
+            AgentConfig("/path/to/config.json")
+    
+    def test_agent_config_file_prefix_valid(self):
+        """Test that file:// prefix is properly handled."""
+        import tempfile
+        import json
+        
+        # Create a temporary config file
+        config_data = {"model": "test-model", "prompt": "Test prompt"}
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.json', delete=False) as f:
+            json.dump(config_data, f)
+            temp_path = f.name
+        
+        try:
+            config = AgentConfig(f"file://{temp_path}", tool_pool=ToolPool())
+            assert config.model == "test-model"
+            assert config.system_prompt == "Test prompt"
+        finally:
+            import os
+            os.unlink(temp_path)
+    
+    def test_to_agent_method_exists(self):
```

Comment:
Can we have `delete=True` (or omit), and include the test in the scope of the context manager here? That way we dont need to do `os.unlink`
```suggestion
        with tempfile.NamedTemporaryFile(mode='w', suffix='.json') as f:
            json.dump(config_data, f)
            config = AgentConfig(f"file://{f.name}", tool_pool=ToolPool())
            assert config.model == "test-model"
            assert config.system_prompt == "Test prompt"
```